src/utils/string/StringUtils.js

```diff
@@ -240,6 +242,117 @@ const getAnonymousChatName = async (members) => {
   });
 };
 
+const helperGetMemberName = (member) => {
+  const isAnonymous = Boolean(member?.anon_user_info_emoji_name);
+
+  return isAnonymous
+    ? `Anonymous ${member?.anon_user_info_emoji_name}`
+    : member?.user?.username ?? member?.user?.name;
+};
+
+const helperIsSelf = (userId, selfSignUserId, selfAnonUserId) => {
+  return userId === selfSignUserId || userId === selfAnonUserId;
+};
+
+const helperGetWhichMembers = (channel, selfSignUserId, selfAnonUserId) => {
+  const members = [];
+  if (channel?.better_channel_member) {
+    if (isArray(channel?.better_channel_member)) {
+      _.forEach(channel?.better_channel_member, (member) => {
+        const isNotSelf = !helperIsSelf(member?.user_id, selfSignUserId, selfAnonUserId);
+        if (isNotSelf) {
+          members.push(member);
+        }
+      });
+
+      return members;
+    }
+
+    _.forIn(channel?.better_channel_member, (member, key) => {
+      const isNotSelf = !helperIsSelf(key, selfSignUserId, selfAnonUserId);
+      if (isNotSelf) {
+        members.push(member);
+      }
+    });
+
+    return members;
+  }
+
+  channel?.members?.forEach((item) => {
+    const isNotSelf = !helperIsSelf(item?.user_id, selfSignUserId, selfAnonUserId);
+    if (isNotSelf) {
+      members.push(item);
+    }
+  });
+
+  return members;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `helperGetWhichMembers` checks if `channel.better_channel_member` is an array or an object and then iterates over it accordingly. This could be simplified by always treating it as an iterable, which would reduce the complexity of the function. However, without knowing the exact data structures used in your application, I can't provide a specific fix.



<!-- This is an auto-generated comment by OSS CodeRabbit -->